wp-content/plugins/trx_utils/importer/export/theme_options.txt
a:192:{s:30:"jacqueline_emailer_subscribers";a:1:{s:21:"E-mailer subscription";a:19:{s:16:"alyana13@mail.ru";s:0:"";s:25:"parkerbenson247@yahoo.com";s:32:"9eee6f96e049c9464d9aac1a4cd081cf";s:20:"web@medialive.com.br";s:32:"28749218eb36859cc15571087a1b4e98";s:25:"ayaz.ahmed@eplanetcom.com";s:32:"47ee74135eec8af24263fe41baa6b1e7";s:27:"lamaisonduyoga.hk@gmail.com";s:32:"a5590792b1c5c372a197227f48c132bd";s:12:"fgg@mail.com";s:32:"39f415257254dd765cd2f469d7f1f28d";s:11:"mel@mel.com";s:32:"fd6ce0f5d282567f637a8a72cebb87c3";s:16:"melissa@test.com";s:32:"13d8ffca04b0fd81264782ad92dca946";s:22:"kanickaraj21@gmail.com";s:0:"";s:25:"luizlopes85spam@gmail.com";s:32:"23e62727d76d92bdbf19c17979b226ee";s:21:"jana.lutman@gmail.com";s:32:"110992d785fd1e19578ea02d26830f44";s:15:"admin@admin.com";s:32:"e25dd6f611f70590e243ad9987e36a8a";s:13:"pin@gmail.com";s:32:"713d8600ce924ad764a649343afc00a1";s:14:"pink@gmail.com";s:32:"11812dd391361829ecccd03fb63072c1";s:19:"honza.sel@gmail.com";s:32:"52bea8cfb49cf0fee54d1e274fe87adc";s:13:"test@test.com";s:32:"6f385536f9795a9d55a858c4fdad2d4c";s:18:"tenro7@hotmail.com";s:32:"0f4faf3081316cf02e6373c186d3c98b";s:18:"avila704@gmail.com";s:0:"";s:18:"gyhkig@dfhbdrf.com";s:32:"035750d80cc35431bb9835fb59c34f2b";}}s:35:"jacqueline_jacqueline_options_reset";s:1:"1";s:18:"jacqueline_options";a:180:{s:10:"body_style";s:4:"wide";s:13:"body_paddings";s:3:"yes";s:10:"theme_skin";s:4:"less";s:11:"body_scheme";s:8:"original";s:11:"body_filled";s:3:"yes";s:9:"bg_custom";s:2:"no";s:8:"bg_color";s:7:"#ffffff";s:10:"bg_pattern";s:0:"";s:17:"bg_pattern_custom";s:0:"";s:8:"bg_image";s:0:"";s:15:"bg_image_custom";s:0:"";s:24:"bg_image_custom_position";s:8:"left_top";s:13:"bg_image_load";s:5:"boxed";s:15:"top_panel_style";s:8:"header_3";s:15:"top_panel_image";s:0:"";s:18:"top_panel_position";s:5:"above";s:16:"top_panel_scheme";s:8:"original";s:18:"pushy_panel_scheme";s:4:"dark";s:15:"show_page_title";s:2:"no";s:16:"show_breadcrumbs";s:2:"no";s:21:"breadcrumbs_max_level";s:1:"0";s:9:"menu_main";s:7:"default";s:15:"menu_attachment";s:5:"fixed";s:11:"menu_slider";s:3:"yes";s:17:"menu_animation_in";s:8:"fadeInUp";s:18:"menu_animation_out";s:11:"fadeOutDown";s:11:"menu_mobile";s:4:"1024";s:10:"menu_width";s:0:"";s:18:"show_top_panel_top";s:3:"yes";s:9:"menu_user";s:7:"default";s:14:"show_languages";s:3:"yes";s:10:"show_login";s:3:"yes";s:14:"show_bookmarks";s:3:"yes";s:12:"show_socials";s:3:"yes";s:4:"logo";s:61:"http://clear30/wp-content/uploads/2018/07/Jacqueline-logo.png";s:11:"logo_retina";s:0:"";s:10:"logo_fixed";s:0:"";s:9:"logo_text";s:0:"";s:11:"logo_height";s:0:"";s:11:"logo_offset";s:0:"";s:11:"show_slider";s:2:"no";s:14:"slider_display";s:8:"fullwide";s:13:"slider_height";s:0:"";s:13:"slider_engine";s:4:"revo";s:12:"slider_alias";s:0:"";s:19:"slider_over_content";s:0:"";s:18:"slider_over_scheme";s:4:"dark";s:15:"slider_category";s:0:"";s:12:"slider_posts";s:1:"5";s:14:"slider_orderby";s:4:"date";s:12:"slider_order";s:4:"desc";s:15:"slider_interval";s:4:"7000";s:17:"slider_pagination";s:2:"no";s:14:"slider_infobox";s:5:"slide";s:20:"slider_info_category";s:3:"yes";s:19:"slider_info_reviews";s:3:"yes";s:24:"slider_info_descriptions";s:0:"";s:15:"custom_sidebars";a:1:{i:0;s:11:"Faq Sidebar";}s:17:"show_sidebar_main";s:5:"right";s:19:"sidebar_main_scheme";s:8:"original";s:12:"sidebar_main";s:12:"sidebar_main";s:19:"show_sidebar_footer";s:3:"yes";s:21:"sidebar_footer_scheme";s:8:"original";s:14:"sidebar_footer";s:14:"sidebar_footer";s:22:"sidebar_footer_columns";s:1:"4";s:23:"show_contacts_in_footer";s:2:"no";s:15:"contacts_scheme";s:8:"original";s:11:"logo_footer";s:0:"";s:18:"logo_footer_retina";s:0:"";s:18:"logo_footer_height";s:2:"30";s:24:"show_copyright_in_footer";s:7:"socials";s:16:"copyright_scheme";s:8:"original";s:11:"menu_footer";s:7:"default";s:16:"footer_copyright";s:269:"<p><a href="http://themerex.net/" target="_blank" rel="noopener">ThemeREX</a> © {Y} All Rights Reserved <a href="http://themerex.net/terms-of-service/" target="_blank" rel="noopener">Terms of Use</a> and <a href="/privacy-policy/" rel="noopener">Privacy Policy</a></p>";s:13:"scroll_to_top";s:3:"yes";s:21:"show_theme_customizer";s:2:"no";s:15:"customizer_demo";s:1:"0";s:13:"css_animation";s:3:"yes";s:19:"animation_on_mobile";s:3:"yes";s:26:"remember_visitors_settings";s:2:"no";s:18:"responsive_layouts";s:3:"yes";s:14:"page_preloader";s:0:"";s:23:"page_preloader_bg_color";s:7:"#ffffff";s:25:"page_preloader_text_color";s:7:"#c1c1c1";s:20:"page_preloader_image";s:0:"";s:12:"privacy_text";s:61:"I agree that my submitted data is being collected and stored.";s:12:"fonts_subset";s:15:"latin,latin-ext";s:15:"custom_css_html";s:2:"no";s:8:"gtm_code";s:0:"";s:9:"gtm_code2";s:0:"";s:11:"custom_code";s:0:"";s:10:"custom_css";s:0:"";s:10:"blog_style";s:7:"excerpt";s:9:"hover_dir";s:13:"left_to_right";s:13:"article_style";s:7:"stretch";s:18:"dedicated_location";s:7:"default";s:12:"show_filters";s:4:"hide";s:9:"blog_sort";s:4:"date";s:10:"blog_order";s:4:"desc";s:14:"posts_per_page";s:2:"12";s:22:"post_excerpt_maxlength";s:3:"250";s:30:"post_excerpt_maxlength_masonry";s:3:"150";s:12:"single_style";s:15:"single-standard";s:4:"icon";s:0:"";s:16:"alter_thumb_size";s:3:"1_1";s:19:"show_featured_image";s:3:"yes";s:15:"show_post_title";s:3:"yes";s:25:"show_post_title_on_quotes";s:2:"no";s:14:"show_post_info";s:3:"yes";s:25:"show_text_before_readmore";s:3:"yes";s:16:"show_post_author";s:3:"yes";s:14:"show_post_tags";s:3:"yes";s:17:"show_post_related";s:3:"yes";s:18:"post_related_count";s:1:"2";s:20:"post_related_columns";s:1:"2";s:17:"post_related_sort";s:4:"date";s:18:"post_related_order";s:4:"desc";s:18:"show_post_comments";s:3:"yes";s:12:"exclude_cats";s:0:"";s:15:"blog_pagination";s:5:"pages";s:13:"blog_counters";s:5:"views";s:14:"close_category";s:8:"parental";s:15:"show_date_after";s:2:"30";s:12:"retina_ready";s:1:"1";s:14:"images_quality";s:3:"100";s:18:"substitute_gallery";s:2:"no";s:21:"gallery_instead_image";s:3:"yes";s:18:"gallery_max_slides";s:1:"5";s:12:"popup_engine";s:8:"magnific";s:16:"substitute_audio";s:3:"yes";s:16:"substitute_video";s:3:"yes";s:16:"use_mediaelement";s:3:"yes";s:12:"social_icons";a:4:{i:0;a:2:{s:3:"url";s:32:"https://twitter.com/ThemeREX_net";s:4:"icon";s:12:"icon-twitter";}i:1;a:2:{s:3:"url";s:42:"https://plus.google.com/+ThemeREXWebStudio";s:4:"icon";s:10:"icon-gplus";}i:2;a:2:{s:3:"url";s:45:"https://business.facebook.com/ThemeRexStudio/";s:4:"icon";s:13:"icon-facebook";}i:3;a:2:{s:3:"url";s:39:"https://www.instagram.com/themerex_net/";s:4:"icon";s:14:"icon-instagram";}}s:10:"show_share";s:10:"horizontal";s:19:"show_share_counters";s:3:"yes";s:13:"share_caption";s:6:"Share:";s:13:"share_buttons";a:1:{i:0;a:2:{s:3:"url";s:0:"";s:4:"icon";s:0:"";}}s:16:"twitter_username";s:0:"";s:20:"twitter_consumer_key";s:0:"";s:23:"twitter_consumer_secret";s:0:"";s:17:"twitter_token_key";s:0:"";s:20:"twitter_token_secret";s:0:"";s:12:"social_login";s:24:"[wordpress_social_login]";s:18:"contact_open_hours";s:24:"Mon - Fry  10:00 - 18:00";s:13:"contact_email";s:17:"info@yoursite.com";s:17:"contact_address_1";s:0:"";s:17:"contact_address_2";s:0:"";s:13:"contact_phone";s:16:"+1 800 245 39 25";s:11:"contact_fax";s:16:"+1 800 245 39 25";s:26:"message_maxlength_contacts";s:4:"1000";s:26:"message_maxlength_comments";s:4:"1000";s:13:"mail_function";s:7:"wp_mail";s:11:"show_search";s:3:"yes";s:15:"use_ajax_search";s:3:"yes";s:22:"ajax_search_min_length";s:1:"4";s:17:"ajax_search_delay";s:3:"500";s:17:"ajax_search_types";s:0:"";s:23:"ajax_search_posts_count";s:1:"5";s:23:"ajax_search_posts_image";s:3:"yes";s:22:"ajax_search_posts_date";s:3:"yes";s:24:"ajax_search_posts_author";s:3:"yes";s:26:"ajax_search_posts_counters";s:3:"yes";s:9:"shop_mode";s:6:"thumbs";s:17:"show_mode_buttons";s:3:"yes";s:17:"shop_loop_columns";s:1:"3";s:13:"show_currency";s:3:"yes";s:9:"show_cart";s:6:"always";s:18:"crop_product_thumb";s:2:"no";s:22:"use_ajax_views_counter";s:2:"no";s:12:"allow_editor";s:2:"no";s:17:"admin_add_filters";s:2:"no";s:25:"show_overriden_taxonomies";s:3:"yes";s:20:"show_overriden_posts";s:3:"yes";s:16:"admin_dummy_data";s:3:"yes";s:19:"admin_dummy_timeout";s:3:"120";s:13:"admin_emailer";s:2:"no";s:17:"admin_po_composer";s:2:"no";s:10:"debug_mode";s:3:"yes";s:10:"api_google";s:39:"AIzaSyDX4e00X7R8eR5eFebM01JXdvFC0AtNV44";s:25:"show_tribe_events_in_blog";s:3:"yes";}s:32:"jacqueline_options_custom_colors";a:3:{s:8:"original";a:34:{s:5:"title";s:8:"Original";s:7:"accent1";s:7:"#F9A392";s:13:"accent1_hover";s:7:"#8ED4CC";s:4:"text";s:7:"#757575";s:10:"text_light";s:7:"#9A9A9A";s:9:"text_dark";s:7:"#323232";s:12:"inverse_text";s:7:"#ffffff";s:13:"inverse_light";s:7:"#B2B2B2";s:12:"inverse_dark";s:7:"#ffffff";s:12:"inverse_link";s:7:"#ffffff";s:13:"inverse_hover";s:7:"#8ED4CC";s:8:"bd_color";s:7:"#F2F2F2";s:8:"bg_color";s:7:"#ffffff";s:8:"bg_image";s:0:"";s:17:"bg_image_position";s:8:"left top";s:15:"bg_image_repeat";s:6:"repeat";s:19:"bg_image_attachment";s:6:"scroll";s:9:"bg_image2";s:0:"";s:18:"bg_image2_position";s:8:"left top";s:16:"bg_image2_repeat";s:6:"repeat";s:20:"bg_image2_attachment";s:6:"scroll";s:10:"alter_text";s:7:"#757575";s:11:"alter_light";s:7:"#9A9A9A";s:10:"alter_dark";s:7:"#232A34";s:10:"alter_link";s:7:"#F9A392";s:11:"alter_hover";s:7:"#8ED4CC";s:14:"alter_bd_color";s:7:"#F4F4F4";s:14:"alter_bd_hover";s:7:"#9A9A9A";s:14:"alter_bg_color";s:7:"#F4F4F4";s:14:"alter_bg_hover";s:7:"#F4F4F4";s:14:"alter_bg_image";s:0:"";s:23:"alter_bg_image_position";s:8:"left top";s:21:"alter_bg_image_repeat";s:6:"repeat";s:25:"alter_bg_image_attachment";s:6:"scroll";}s:5:"light";a:34:{s:5:"title";s:5:"Light";s:7:"accent1";s:7:"#20C7CA";s:13:"accent1_hover";s:7:"#189799";s:4:"text";s:7:"#ffffff";s:10:"text_light";s:7:"#ffffff";s:9:"text_dark";s:7:"#ffffff";s:12:"inverse_text";s:7:"#ffffff";s:13:"inverse_light";s:7:"#ffffff";s:12:"inverse_dark";s:7:"#ffffff";s:12:"inverse_link";s:7:"#ffffff";s:13:"inverse_hover";s:7:"#ffffff";s:8:"bd_color";s:7:"#dddddd";s:8:"bg_color";s:7:"#8ED4CC";s:8:"bg_image";s:0:"";s:17:"bg_image_position";s:8:"left top";s:15:"bg_image_repeat";s:6:"repeat";s:19:"bg_image_attachment";s:6:"scroll";s:9:"bg_image2";s:0:"";s:18:"bg_image2_position";s:8:"left top";s:16:"bg_image2_repeat";s:6:"repeat";s:20:"bg_image2_attachment";s:6:"scroll";s:10:"alter_text";s:7:"#8a8a8a";s:11:"alter_light";s:7:"#acb4b6";s:10:"alter_dark";s:7:"#232a34";s:10:"alter_link";s:7:"#20c7ca";s:11:"alter_hover";s:7:"#189799";s:14:"alter_bd_color";s:7:"#e7e7e7";s:14:"alter_bd_hover";s:7:"#dddddd";s:14:"alter_bg_color";s:7:"#ffffff";s:14:"alter_bg_hover";s:7:"#f0f0f0";s:14:"alter_bg_image";s:0:"";s:23:"alter_bg_image_position";s:8:"left top";s:21:"alter_bg_image_repeat";s:6:"repeat";s:25:"alter_bg_image_attachment";s:6:"scroll";}s:4:"dark";a:34:{s:5:"title";s:4:"Dark";s:7:"accent1";s:7:"#F9A392";s:13:"accent1_hover";s:7:"#8ED4CC";s:4:"text";s:7:"#757575";s:10:"text_light";s:7:"#9A9A9A";s:9:"text_dark";s:7:"#323232";s:12:"inverse_text";s:7:"#FFFFFF";s:13:"inverse_light";s:7:"#B2B2B2";s:12:"inverse_dark";s:7:"#ffffff";s:12:"inverse_link";s:7:"#ffffff";s:13:"inverse_hover";s:7:"#8ED4CC";s:8:"bd_color";s:24:"rgba(242, 242, 242, 0.6)";s:8:"bg_color";s:7:"#FFFFFF";s:8:"bg_image";s:0:"";s:17:"bg_image_position";s:8:"left top";s:15:"bg_image_repeat";s:6:"repeat";s:19:"bg_image_attachment";s:6:"scroll";s:9:"bg_image2";s:0:"";s:18:"bg_image2_position";s:8:"left top";s:16:"bg_image2_repeat";s:6:"repeat";s:20:"bg_image2_attachment";s:6:"scroll";s:10:"alter_text";s:7:"#757

wp-content/plugins/pegasaas-accelerator-wp/assets/js/settings.js
var pending_form = "";	
jQuery(".feature-switch,.feature-setting-change,.prompt-to-clear-cache").bind("submit", function(e) {
	
	if (jQuery(this).parents(".pegasaas-feature-box").hasClass("locked-for-novice") && jQuery(".pegasaas-dashboard").hasClass("interface-novice")) {
		e.preventDefault();
		return false;
	} else if (jQuery(this).parents(".feature-container").hasClass("locked-for-novice") && jQuery(".pegasaas-dashboard").hasClass("interface-novice")) {
		e.preventDefault();
		return false;
	}
	
	// change the parent option to show ENABLED or DISABLED state
	var parent_id = jQuery(this).parents(".settings-panel").attr("id");
	var current_state = jQuery(this).find("input[name='c']").val();
	var span = jQuery("[data-target='#" + parent_id + "']").find("span.label-state");
	if (current_state == "enable-feature") {
		span.html("ENABLED");
		span.removeClass("label-disabled");
		span.addClass("label-enabled");
		
	} else if (current_state == "disable-feature") {
		span.html("DISABLED");
		span.removeClass("label-enabled");
		span.addClass("label-disabled");
	}

	
	
	if (jQuery(this).find("input[name='prompt']").length > 0) {
		
		if (jQuery(this).find("input[name='prompt']").val() == "clear_cache") {
			e.preventDefault();
			pending_form = jQuery(this);


			jQuery("#confirm-update-clear-cache").modal("show");
			jQuery("#update-setting-and-clear").attr("data-feature", jQuery(this).find("input[name='f']").val());


			jQuery("#update-setting-and-clear").bind("click", function() {
				//alert(jQuery(pending_form).find("input[name='c']").val());
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status"  || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting" ||
				    jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting" ||
				    jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting"  ) {

					
					
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}
				}
				

				jQuery(pending_form).find("input[name='cache']").remove();
				jQuery(this).unbind("click");

				submit_via_ajax(pending_form, "pegasaas_dashboard_settings_update");
				
				//jQuery(pending_form).submit();
				jQuery("#confirm-update-clear-cache").modal("hide");
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting") {
					if (jQuery(pending_form).parents(".feature-row").hasClass("feature-disabled")) {
						jQuery(pending_form).parents(".feature-row").removeClass("feature-disabled");
					} else {
						jQuery(pending_form).parents(".feature-row").addClass("feature-disabled");
					}					
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}	
				}
			});

			
			jQuery("#update-setting-but-no-clear").bind("click", function() {
				
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting"|| 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status"  ||
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting"||
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting"|| 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting") {
								
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}
				}
				
				
				
				if (jQuery(pending_form).find("input[name='cache']").length == 0) {
					jQuery(pending_form).append("<input type='hidden' name='cache' value='do-not-clear' />");

				}

				submit_via_ajax(pending_form, "pegasaas_dashboard_settings_update");
				//jQuery(pending_form).submit();
				jQuery(this).unbind("click");
				jQuery("#confirm-update-clear-cache").modal("hide");
				
				
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting"|| 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting") {

				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}	
				}
				
				
				console.log("after save 2, state is now: " + jQuery(pending_form).find("input[name='c']").val());


			});	
		
		
		} else if (jQuery(this).find("input[name='prompt']").val() == "clear_image_cache") {
			e.preventDefault();
			pending_form = jQuery(this);
			


			jQuery("#confirm-update-clear-image-cache").modal("show");
			jQuery("#update-setting-and-clear-images").attr("data-feature", jQuery(this).find("input[name='f']").val());


			jQuery("#update-setting-and-clear-images").bind("click", function() {
				
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status"  || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting" ||
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
				    jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting" ||
				    jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting"  ) {

					
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}
				}
				

				
				jQuery(pending_form).find("input[name='image_cache']").remove();
				jQuery(this).unbind("click");

				//jQuery(pending_form).submit();
				submit_via_ajax(pending_form, "pegasaas_dashboard_settings_update");
				jQuery("#confirm-update-clear-image-cache").modal("hide");
				
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting") {
					
					if (jQuery(pending_form).parents(".feature-row").hasClass("feature-disabled")) {
						jQuery(pending_form).parents(".feature-row").removeClass("feature-disabled");
					} else {
						jQuery(pending_form).parents(".feature-row").addClass("feature-disabled");
					}	
					
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}	
				}
			});

			
			jQuery("#update-setting-but-no-clear-images").bind("click", function() {
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting"|| 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status"  ||
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting"||
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting") {
									
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}
				}
				
				
				if (jQuery(pending_form).find("input[name='image_cache']").length == 0) {
					jQuery(pending_form).append("<input type='hidden' name='image_cache' value='do-not-clear' />");

				}
				jQuery(this).unbind("click");
				//jQuery(pending_form).submit();
				submit_via_ajax(pending_form, "pegasaas_dashboard_settings_update");
		jQuery("#confirm-update-clear-image-cache").modal("hide");
				if (jQuery(pending_form).find("input[name='c']").val() == "change-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-status" || 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-setting" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-feature-attribute" || 
					jQuery(pending_form).find("input[name='c']").val() == "change-local-complex-setting"|| 
					jQuery(pending_form).find("input[name='c']").val() == "toggle-local-complex-setting") {

						
				} else {
					if (jQuery(pending_form).find("input[name='c']").val() == "enable-feature") {
						jQuery(pending_form).find("input[name='c']").val("disable-feature");
					} else {
						jQuery(pending_form).find("input[name='c']").val("enable-feature");
					}	
				}


			}	);	
		
		} else if (jQuery(this).find("input[name='prompt']").val() == "no-prompt-image-cache") {
			
			jQuery(this).find("input[name='prompt']").val("clear_image_cache");
			return true;
		} else {
			
			jQuery(this).find("input[name='prompt']").val("clear_cache");
			return true;
		}

	} else {
		e.preventDefault();
		// there is no prompt, so just submit the form
		
		jQuery(this).find("input[name='prompt']").val("clear_cache");
	
		submit_via_ajax(this, "pegasaas_dashboard_settings_update");
		
		
		return true;
	}
});


function submit_form(element) {
	jQuery(element).parents('form').attr("target", "hidden-frame")
	jQuery(element).parents('form').submit();
}
	
jQuery(".table-complex-settings select").bind("change", function () {
	jQuery(this).parents("form.feature-toggle-switch").find("input[name=c]").val("change-local-complex-setting");
	jQuery(this).parents("form.feature-toggle-switch").submit();
});
jQuery(".table-complex-settings form.feature-toggle-switch .js-switch").bind("change", function(e) { 
	
		jQuery(this).parents("form.feature-toggle-switch").find("input[name=c]").val("toggle-local-complex-setting");

});

function submit_via_ajax(the_form, action) {
	var the_args = get_form_data(jQuery(the_form));
		jQuery(the_form).find("button").attr("disabled", true);
	
		var save_button = jQuery(the_form).find(".btn-save");
		if (save_button) {
			save_button.attr("disabled", true);
			save_button.find("span.status").html("<i class='svg-icons svg-icon-14 svg-tail-spin-white'></i>");
		}
		
		jQuery.post({url: ajaxurl, 
				 type: "POST",
				 dataType: "json",
				 data: { 'action': action, 'api_key': jQuery("#pegasaas-api-key").val(), 'args': the_args},
				 
				 success: function(data) {
					 if (save_button) {
						save_button.removeAttr("disabled");
						save_button.find("span.status").html("");
					}
					
					 if (the_args['c'] == 'change-feature-attribute' && the_args['f'] == 'coverage' && data['cache_clearing_queued']) {
						
						 run_background_cache_clearing();
					 }
					//if (the_args['c'] == "remove-item-complex-setting" && the_args['f'] == 'multi_server_ip') {
					//	jQuery($this).parents("li").remove();
					//}
					 
				},
				 error: function(jqXHR, textStatus, errorThrown) {
				  alert("Error, status = " + textStatus + ", " +
              "error thrown: " + errorThrown);
			}
				 });	
	
}
	
	jQuery(".background-submit").submit(function(event) {
		event.preventDefault();
	
		
		var $this = this;
		var the_args = get_form_data(jQuery(this));
		jQuery(this).find("button").attr("disabled", true);
		
		jQuery.post({url: ajaxurl, 
				 type: "POST",
				 dataType: "json",
				 data: { 'action': 'pegasaas_dashboard_settings_update', 'api_key': jQuery("#pegasaas-api-key").val(), 'args': the_args},
				 
				 success: function(data) {
					if (the_args['c'] == "remove-item-complex-setting" && the_args['f'] == 'multi_server_ip') {
						jQuery($this).parents("li").remove();
					}
					 
				},
				 error: function(jqXHR, textStatus, errorThrown) {
				  alert("Error, status = " + textStatus + ", " +
              "error thrown: " + errorThrown
        );
				 }
				 });
		
	});
	
	function get_form_data(form){
		var unindexed_array = form.serializeArray();
		var indexed_array = {};

		jQuery.map(unindexed_array, function(n, i){
			indexed_array[n['name']] = n['value'];
		});

		return indexed_array;
	}	

// side menu click action
jQuery(".quicklinks-column ul a").bind("click", function(e) {
	e.preventDefault();
	
	var target_id = jQuery(this).attr("href");
	
	jQuery(".settings-panel").css("display", "none");
	jQuery(".pegasaas-feature-section-container").fadeIn();
	
	jQuery('body,html').animate({
		scrollTop: jQuery(target_id).offset().top - 50
		}, 1000
	);
});


// section feature click action						
jQuery(".settings-container a.list-group-item").click(function(e) {
	e.preventDefault();
	var data_target = jQuery(this).attr("data-target");
	jQuery("#pegasaas-accelerator-main-settings .pegasaas-feature-section-container").css("display", "none");
	
	//jQuery('body').scrollspy({ target: '#quicklinks', offset: 60 });
	//
	
	jQuery(data_target).fadeIn();
	

	jQuery("#quicklinks").addClass("in-feature");
});




// back to main section click action
jQuery(".back-to-main-settings").click(function(e) {
	e.preventDefault();
	var data_target = jQuery(this).parents(".settings-panel").attr("id");
	var target = jQuery("[data-target='#" + data_target + "']").parents(".pegasaas-feature-section-container");
	jQuery(".settings-panel").css("display", "none");
	
	jQuery("#pegasaas-accelerator-main-settings .pegasaas-feature-section-container").fadeIn();
	jQuery('[data-spy="scroll"]').each(function () {
  	var $spy = jQuery(this).scrollspy('refresh')
});
	
	jQuery('body,html').animate({ scrollTop: jQuery(target).offset().top - 50 }, 1000);
	//jQuery('body').scrollspy({ target: '#quicklinks', offset: 60 });
	jQuery("#quicklinks").removeClass("in-feature");
	
	
});


// back to main section click action
jQuery(".list-group-item a[data-toggle='collapse']").click(function(e) {
	
	var parent = jQuery(this).parents(".list-group-item");
	
	if (parent.hasClass("open")) {
		parent.removeClass("open");
		jQuery(this).find(".fa").removeClass("fa-angle-up").addClass("fa-angle-down");
	} else {
		parent.addClass("open");
				jQuery(this).find(".fa").removeClass("fa-angle-down").addClass("fa-angle-up");

	}


});

jQuery("#settings-nav-button").on('shown.bs.tab', function(e) {
	 
	jQuery('body').scrollspy({ target: '#quicklinks', offset: 60 });

});